starfish/image/_stack.py

```diff
@@ -1,6 +1,8 @@
 import collections
 import os
 import typing
+from typing import Iterable, Mapping, Iterator, Union
```

Comment:
Agree. I added this in as a subtle suggestion to provoke conversation (It worked! :))

I prefer the way I did it here. My rationale is: 
1. Most important: these typing examples can get really long, particularly in cases with dependent return values (`Union`, `Optional`, etc) which can be annoying to code while considering line length limits. Anything we can do to shorten the signature is good. 
2. Because typing examples are used in prescribed locations, I don't think there is a significant reduction in clarity.